packages/sdk-core/src/BacktraceCoreClient.ts

```diff
@@ -158,19 +165,28 @@ export abstract class BacktraceCoreClient {
         ).build();
 
         if (metrics) {
-            this._metrics = metrics;
+            this._modules.set(BacktraceMetrics, metrics);
         }
 
         if (this.options.breadcrumbs?.enable !== false) {
-            this.breadcrumbsManager = new BreadcrumbsManager(this.options?.breadcrumbs, this._setup.breadcrumbsSetup);
-            this._attachments.push(this.breadcrumbsManager.breadcrumbsStorage);
-            this.attributeManager.addProvider(this.breadcrumbsManager);
+            const breadcrumbsManager = new BreadcrumbsManager(this.options?.breadcrumbs, this._setup.breadcrumbsSetup);
+            this._attachments.push(breadcrumbsManager.breadcrumbsStorage);
+            this.attributeManager.addProvider(breadcrumbsManager);
+            this._modules.set(BreadcrumbsManager, breadcrumbsManager);
         }
 
-        this.initialize();
         this._enabled = true;
     }
 
+    public initialize() {
+        for (const module of this._modules.values()) {
```

Comment:
we talked about this via slack but what is the advantage of this? Actually via map we're hiding what modules we're initializing. The previous version allowed us to clearly says what starts and when. Now we need to check when we're adding it. Backtrace Client isn't less dependent than it was earlier but instead we're hiding something behind the implementation detail.


Why this is better solution if the number of modules previously initialized was seriously small?